File: testbed/test/cruise/associations/UnidirectionalOptionalOneTest.java
package cruise.associations;

import org.junit.*;

import cruise.associations.MentorAI;
import cruise.associations.ProgramAI;
import cruise.associations.StudentAI;

public class UnidirectionalOptionalOneTest
{

  @Test
  public void setStudent()
  {
    StudentAI s = new StudentAI(1);
    MentorAI m = new MentorAI("a");
    Assert.assertEquals("a",m.getName());
    Assert.assertEquals(null,m.getStudent());

    m.setStudent(s);
    Assert.assertEquals(s,m.getStudent());
    Assert.assertEquals(1, m.getStudent().getNumber());
  }
  
  @Test
  public void deleteLeavesStudentAlone()
  {
    MentorAI m = new MentorAI("a");
    StudentAI s = new StudentAI(1);
    m.setStudent(s);
    ProgramAI p = new ProgramAI();
    s.setProgram(p);
    m.delete();
    Assert.assertEquals(null, m.getStudent());
    Assert.assertEquals(null, m.getProgram());
    Assert.assertEquals(p, s.getProgram());
    
  }
  
}

File: testbed_python/test/associations/unidirectional_optional_one_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorAI", "ProgramAI", "StudentAI"], ["associations"])
from ImportModules import *

class UnidirectionalOptionalOneTest(unittest.TestCase):
    def test_setStudent(self):
        s = StudentAI.StudentAI(1)
        m = MentorAI.MentorAI("a")
        self.assertEqual("a",m.getName())
        #self.assertEqual(None, m.getStudent())

        m.setStudent(s)
        self.assertEqual(s,m.getStudent())
        self.assertEqual(1, m.getStudent().getNumber())

    def test_deleteLeavesStudentAlone(self):
        m = MentorAI.MentorAI("a")
        s = StudentAI.StudentAI(1)
        m.setStudent(s)
        p = ProgramAI.ProgramAI()
        s.setProgram(p)

        del m
        self.assertEqual(p, s.getProgram())
